ios/CoinbaseWalletSDK/Task/TaskManager.swift

```diff
@@ -35,16 +37,28 @@ class TaskManager {
         return true
     }
     
-    func findRequest(for response: EncryptedResponseMessage) -> RequestMessage? {
-        guard let task = tasks[response.content.requestId] else {
+    static func findRequest(for requestId: UUID) -> RequestMessage? {
+        guard let task = tasks[requestId] else {
             return nil
         }
     
         return task.request
     }
     
-    func reset() {
-        tasks.removeAll()
+    static func getHost(for response: EncryptedResponseMessage) -> URL? {
+        guard let task = tasks[response.content.requestId] else {
+            return nil
+        }
+    
+        return task.host
+    }
+    
+    static func reset(host: URL) {
+        tasks.forEach { task in
+            if task.value.host == host {
+                tasks.removeValue(forKey: task.key)
+            }
+        }
```

Comment:
we can do something like
```suggestion
        tasks.removesAll(where: { $.value.host == host })
```

Reply:
this doesn't appear to compile

Reply:
@AMITGOELNY 
tasks.removesAll(where: { $0.value.host == host })
https://developer.apple.com/documentation/swift/array/removeall(where:)-5k61r

Reply:
i think you can figure out how to make it compile